Shared/M365/GraphConnection.ps1

```diff
@@ -0,0 +1,178 @@
+﻿# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+<#
+.SYNOPSIS
+This script defines a function `Connect-GraphAdvanced` that establishes a connection to Microsoft Graph.
+It ensures that the required modules are installed and loaded.
+The function accepts a list of scopes and modules, with optional parameters for tenant ID and connection details display.
+If the required modules are not found, the script attempts to install them.
+The function returns the connection information or null if the connection fails.
+
+.PARAMETER Scopes
+ Mandatory array of strings specifying the scopes for the connection.
+.PARAMETER Modules
+ Mandatory array of strings specifying the modules required for the connection.
+.PARAMETER TenantId
+ Optional array of strings specifying the tenant ID(s) for the connection.
+.PARAMETER DoNotShowConnectionDetails
+ Optional switch to hide connection details.
+.PARAMETER MinModuleVersion
+ Optional parameter to specify the minimum version of the Graph modules (default is 2.0.0).
+
+.OUTPUTS
+Microsoft.Graph.PowerShell.Authentication.AuthContext. The connection information object for the Microsoft Graph session.
+
+.EXAMPLE
+$graphConnection = Connect-GraphAdvanced -Scopes User.Read, Mail.Read -Modules Microsoft.Graph
+This example establishes a connection to Microsoft Graph with the scopes "User.Read" and "Mail.Read" using the "Microsoft.Graph" module.
+
+.EXAMPLE
+$graphConnection = Connect-GraphAdvanced -Scopes Group.Read.All, User.Read.All -Modules Microsoft.Graph.Users, Microsoft.Graph.Groups
+This example establishes a connection to Microsoft Graph with the scopes "Group.Read.All" and "User.Read.All" using the "Microsoft.Graph.Users" and "Microsoft.Graph.Groups" modules.
+
+.EXAMPLE
+$graphConnection = Connect-GraphAdvanced -Scopes Group.Read.All, User.Read.All -Modules Microsoft.Graph.Users, Microsoft.Graph.Groups -minModuleVersion 2.25.0
+This example establishes a connection to Microsoft Graph with the scopes "Group.Read.All" and "User.Read.All" using the "Microsoft.Graph.Users" and "Microsoft.Graph.Groups" modules, and specifies a minimum module version of 2.25.0.
+#>
+
+. $PSScriptRoot\..\ModuleHandle.ps1
+
+function Connect-GraphAdvanced {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string[]]$Scopes,
+        [Parameter(Mandatory = $true)]
+        [string[]]$Modules,
+        [Parameter(Mandatory = $false)]
+        [string[]]$TenantId = $null,
+        [Parameter(Mandatory = $false)]
+        [switch]$DoNotShowConnectionDetails,
+        [Parameter(Mandatory = $false)]
+        [System.Version]$MinModuleVersion = 2.0.0.0
```

Comment:
Let's add a script validation here for the min version supported.

Let's also add some version information to get the module that we are currently using, just in case things change with Graph.